src/openms/source/CHEMISTRY/AASequence.cpp

```diff
@@ -1377,9 +1380,14 @@ namespace OpenMS
   void AASequence::parseString_(const String& pep, AASequence& aas,
                                 bool permissive)
   {
+    // Reserving space, populate it and then shrink again (since we probably
+    // over-allocate due to modifications). This substantially speeds up the
+    // function for unmodified sequences (3x speedup).
     aas.peptide_.clear();
+
     String peptide(pep);
     peptide.trim();
```

Comment:
copying the string and trimming it seems also kind of a waste here. but probably not too much time spend here